### PR TITLE
Cli ethan

### PR DESCRIPTION
**Pull request by ehn4602**
Implemented CLI Functionality for play, search_user, add_friend, remove_friend, search_game, and help
functions modified  in IAPP  old {search_friend} -> new {search_friends}
function added to IAPP {search_users}

Important notes: 
1. Sort by genre is outdated since genre in this branch is a string not an array, Let me know if this should be fixed before merge which will cause additional merge conflicts.
2. help command information is a placeholder and will be changed based on feedback

### CODE REVIEW

File: src/app/cli/CommandLineInterface.java [hunk 1]
```diff
@@ -520,6 +541,295 @@ public void launch(){
                     rate(tokens[1], tokens[2]);
                 }//add more cases
 
+                case PLAY -> {
+                    if(tokens.length <3){
+                        System.out.println("\nUsage:");
+                        System.out.println("PLAY { game | time }");
```
Comment (cecilialau6776) [inline src/app/cli/CommandLineInterface.java:840]: common usage messages have the format of `<required parameter> [optional parameter]`. not very important but worth noting at the very least if not changing at some point

File: src/app/cli/CommandLineInterface.java [hunk 2]
```diff
@@ -520,6 +541,295 @@ public void launch(){
...
+                        System.out.println("There is not a game with that name.");
+                        continue;
+                    }
+
+                    Game selected_game = game_list[0];
+                    if(game_list.length > 1) {
+                        System.out.println("Which game would you like to play (enter the number)?");
+                        for(int i = 0; i < game_list.length; ++i) {
+                            Game curr_game = game_list[i];
+                            System.out.println(i+1 + ".\tName: " + curr_game.title() + "\n\tPlay time: " +
```
Comment (cecilialau6776) [inline src/app/cli/CommandLineInterface.java]: should be put into a `toString`

File: src/app/cli/CommandLineInterface.java [hunk 3]
```diff
@@ -520,6 +541,295 @@ public void launch(){
...
+                        System.out.println("email           email that the account is associated with to be searched\n");
+                        continue;
+                    }
+                    User[] user_list = app.search_users(tokens[1]);
+
+                    if(user_list.length==0) {
+                        System.out.println("There no users linked to this email address");
+                        continue;
+                    }
+                    if(user_list.length > 1) {
```
Comment (cecilialau6776) [inline src/app/cli/CommandLineInterface.java]: the case where `user_list.length == 1` is missed; consider just moving the code inside this `if` block outside of it and removing the `if` itself

File: src/app/cli/CommandLineInterface.java [hunk 4]
```diff
@@ -520,6 +541,295 @@ public void launch(){
...
+                        System.out.println("SEARCH_GAME { search_val | search_type | sort_val | descend }");
+                        System.out.println("search_val      value being searched");
+                        System.out.println("search_type     type that the being search is, below are acceptable inputs");
+                        System.out.println("                {title},{platform},{date},{developer},{price},{genre}");
+                        System.out.println("sort_val        optional argument to sort the results by, below are acceptable inputs");
+                        System.out.println("                {title},{price},{genre},{developer},{date}");
+                        System.out.println("descend         makes sort into a descend if val=\"D\" otherwise stays ascended\n");
+                        continue;
+                    }
+                    Comparator<Game> default_comparator = new Comparator<Game>() {
```
Comment (cecilialau6776) [inline src/app/cli/CommandLineInterface.java]: comparators should be moved to `Game.java`, i'll do this myself at some point

File: src/app/cli/CommandLineInterface.java [hunk 5]
```diff
@@ -520,6 +541,295 @@ public void launch(){
...
+                            } else {
+                                return result;
+                            }
+                        }
+                    };
+                    Comparator<Game> title_comparator = new Comparator<Game>() {
+                        @Override
+                        public int compare(Game o1, Game o2) {
+                            int result;
+                            if((tokens.length>3)&&(tokens[4].equals("D"))) {
```
Comment (cecilialau6776) [inline src/app/cli/CommandLineInterface.java]: dont do this check in the comparator, you can easily reverse a comparator [like this](https://stackoverflow.com/questions/32995559/reverse-a-comparator-in-java-8)

File: src/app/cli/CommandLineInterface.java
Comment (cecilialau6776) [inline src/app/cli/CommandLineInterface.java:1]: like with anastasia's, you should move all of the switch cases into their own functions

**Review by cecilialau6776 — CHANGES_REQUESTED** [review]

File: src/app/cli/CommandLineInterface.java (hunk 2 above)
Comment (ehn4602) [reply src/app/cli/CommandLineInterface.java]: Do you want the game to be printed using it's toString? or make a toString to then use it?

File: src/app/cli/CommandLineInterface.java (hunk 1 above)
Comment (ehn4602) [reply src/app/cli/CommandLineInterface.java:840]: Yea sounds good, I'll remake the usage into a better format

File: src/app/cli/CommandLineInterface.java (hunk 4 above)
Comment (ehn4602) [reply src/app/cli/CommandLineInterface.java]: Ok, I still plan to fix the comparators in the CLI to allow the reversing to be implemented as stated in a different Comment.

File: src/app/cli/CommandLineInterface.java (hunk 2 above)
Comment (cecilialau6776) [reply src/app/cli/CommandLineInterface.java]: i dont understand
you could just do what anastasia did and just write a function that prints a `Game`

**Review by cecilialau6776 — APPROVED** [review]

**Pull request merged**